### PR TITLE
Make Painless static state more than 2x smaller saving 7M+ of heap at idle

### PR DESCRIPTION
**Pull request by original-brownbear**
First off, deduplicate painless clasess more aggressively across contexts. Also, save the memory for the static deduplication maps by just passing local variables through the build process. This only runs a single time anyway so there's no point in retaining these static maps, we'll never use them again.

Before, unused idle painless:
image

After:

image

12.4M of static state become 4.7M saving almost 8M on every ES node (since painless is a module now)

### CODE REVIEW

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookup.java [hunk 1]
```diff
@@ -163,18 +163,6 @@ public PainlessMethod lookupPainlessMethod(Class<?> targetClass, boolean isStati
         return lookupPainlessObject(targetClass, objectLookup);
     }
 
-    public List<PainlessMethod> lookupPainlessSubClassesMethod(String targetCanonicalClassName, String methodName, int methodArity) {
```
Comment (original-brownbear) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookup.java:166]: Not strictly necessary here but these appear unused so I dropped them here as they're not really worth a separate PR I figured?

Comment (rjernst) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookup.java:166]: @jdconrad Do you know if this method was refactored away and forgotten, or if it is a bug that it is unused?

**Review by rjernst — APPROVED** [review]
The general changes look good to me, but I would like @jdconrad to take a quick look.

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookup.java (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookup.java:166]: These have never been used. There were plans to use these when refactoring portions of the AST, but they never came to fruition. I'm good with removing these now.

**Review by jdconrad — APPROVED** [review]
LGTM!

**Comment by original-brownbear** [comment]
@elasticmachine update branch

**Comment by original-brownbear** [comment]
Thanks you two!

**Pull request merged**